src/LegacyTranslationWriter.php

```diff
@@ -15,20 +15,22 @@
 use Symfony\Component\Translation\Writer\TranslationWriter;
 
 /**
- * This writer is just a legacy wrapper for Symfony TranslationWriter
- * and provide a BC layer for Symfony 4.
+ * This writer is just a wrapper for Symfony TranslationWriter
+ * and provide a BC layer for Symfony 2.7 to 3.3.
  *
  * @author Victor Bocharsky <bocharsky.bw@gmail.com>
  */
-class LegacyTranslationWriter
+final class LegacyTranslationWriter // implements Symfony\Component\Translation\Writer\TranslationWriterInterface
 {
-    /**
-     * @var TranslationWriter
-     */
     private $writer;
 
-    public function __construct(TranslationWriter $writer)
+    public function __construct($writer)
```

Comment:
The same here, I think we can still use `TranslationWriter` type hint. Or do you want to show a better-explained error with throwing `LogicException` in its constructor?

Reply:
Exactly. I removed it here just because I wanted a little better exception message.